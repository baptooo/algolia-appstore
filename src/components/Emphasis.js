import React from 'react';
import SafelySetInnerHTML from 'safely-set-inner-html';
import { withStyles } from 'material-ui/styles';

const instance = new SafelySetInnerHTML({
  ALLOWED_TAGS: ['em'],
  ALLOWED_ATTRIBUTES: ['className']
});

const styles = theme => ({
  emphasis: {
    borderWidth: '0 0 2px 0',
    borderColor: '#00AEFF',
    borderStyle: 'solid',
    padding: '2px 4px'
  }
});

const Emphasis = ({ value, classes }) => (
  <span>{instance.transform(value.replace(/<em/g, `<em className=${classes.emphasis}`))}</span>
);

export default withStyles(styles)(Emphasis);