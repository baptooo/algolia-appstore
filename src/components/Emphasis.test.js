import React from 'react';
import renderer from 'react-test-renderer';
import Emphasis from './Emphasis';

describe('(Component) Emphasis', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Emphasis value={'<em>Foo</em> bar'} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});