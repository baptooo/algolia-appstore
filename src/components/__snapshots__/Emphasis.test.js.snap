// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) Emphasis should render correctly 1`] = `
<span>
  <em
    className="Emphasis-emphasis-1"
  >
    Foo
  </em>
   bar
</span>
`;
